runtime-modules/content-directory/src/lib.rs

```diff
@@ -1032,38 +1031,99 @@ decl_module! {
+
+            // Ensure provided new_entity_controller is not equal to current one
+            entity.get_permissions_ref().ensure_controllers_are_not_equal(&new_controller)?;
 
+            // Ensure any inbound PropertyValue::Reference with same_owner flag set points to the given Entity
             entity.ensure_inbound_same_owner_rc_is_zero()?;
 
+            let class_properties = class.properties;
+
+            let entity_property_values = entity.values;
+
+            // Create wrapper structure from provided entity_property_values and their corresponding Class properties
+            let values_for_existing_properties = ValuesForExistingProperties::from(&class_properties, &entity_property_values);
+
+            // Filter provided values_for_existing_properties, leaving only `Reference`'s with `SameOwner` flag set
+            // Retrieve the set of corresponding property ids
+            let entity_property_id_references_with_same_owner_flag_set =
+                Self::get_property_id_references_with_same_owner_flag_set(values_for_existing_properties);
+
+            // Ensure provided `new_property_value_references_with_same_owner_flag_set` are valid references with `SameOwner` flag set
+            Self::ensure_only_references_with_same_owner_flag_set_provided(
+                &entity_property_id_references_with_same_owner_flag_set,
+                &new_property_value_references_with_same_owner_flag_set
+            )?;
+
+            // Perform checks to ensure all `new_property_value_references_with_same_owner_flag_set`
+            // under provided `entity_property_id_references_with_same_owner_flag_set` are valid
+            Self::ensure_property_values_are_valid(
+                &class_properties, &entity_property_id_references_with_same_owner_flag_set,
+                &new_controller, &new_property_value_references_with_same_owner_flag_set
+            )?;
+
+            // Make updated entity_property_values from parameters provided
+            let entity_property_values_updated =
+                if let Some(entity_property_values_updated) =
+                    Self::make_updated_property_value_references_with_same_owner_flag_set(
+                        entity_property_id_references_with_same_owner_flag_set, &entity_property_values,
+                        &new_property_value_references_with_same_owner_flag_set,
+                    ) {
+
+                        // Create wrapper structure from provided entity_property_values_updated and their corresponding Class properties
+                        let values_for_updated_properties = ValuesForExistingProperties::from(&class_properties, &entity_property_values_updated);
+
+                        // Traverse all values_for_updated_properties to ensure unique property satisfied (if required)
+                        Self::ensure_property_values_unique_option_satisfied(values_for_updated_properties)?;
```

Comment:
As per comment above, why is this on its own outside of `ensure_property_values_are_valid` here?

Reply:
Good point, already decoupled `required` check as well here https://github.com/Joystream/joystream/commit/affc8b4b9b482df85445f3c3ba924fd586f42dee